build.rs

```diff
@@ -21,29 +21,61 @@ struct IdempotentWrapper<'a> {
     body: &'a str,
 }
 
+#[derive(Template)]
+#[template(path = "promscale.control", escape = "none")]
+struct ControlFile {
+    is_pg_12: bool,
+}
+
+const CONTROL_FILE_NAME: &str = "promscale.control";
 const MIGRATION_FILE_NAME: &str = "hand-written-migration.sql";
 
-/// This build script reads the SQL files placed in the `migration/{idempotent,migration}`
-/// directories, wraps each file in its respective template (in the `templates` directory), and
-/// outputs the whole contents as one long SQL file (`MIGRATION_FILE_NAME`) in the root directory.
 fn main() {
     println!("cargo:rerun-if-changed=Cargo.lock");
     // Note: directories are not traversed, instead Cargo looks at the mtime of the directory.
     println!("cargo:rerun-if-changed=migration");
     println!("cargo:rerun-if-changed=migration/migration");
     println!("cargo:rerun-if-changed=migration/idempotent");
     println!("cargo:rerun-if-changed=templates");
+    println!("cargo:rerun-if-changed={}", CONTROL_FILE_NAME);
```

Comment:
This causes `build.rs` to always re-compile, because as specified, this is an output which `build.rs` is producing. Did you mean:

```
println!("cargo:rerun-if-changed=templates/{}", CONTROL_FILE_NAME);
```

Reply:
I suspect that something else weird is going on here. I'm looking into it.